src/flag_gems/ops/cross_entropy_loss.py

```diff
@@ -488,47 +524,77 @@ def forward(ctx, inp, target, weight, reduction, ignore_index, label_smoothing):
         inp = inp.contiguous()
         tgt = target.contiguous()
         weight = weight.contiguous()
-        out = torch.zeros(shape, dtype=torch.float32, device=inp.device)
+        out = torch.empty(shape, dtype=torch.float32, device=inp.device)
         grid = lambda meta: (N, triton.cdiv(D, meta["BLOCK_D"]))
 
         if tgt.ndim == dim:
             # target probabilities
             with torch.cuda.device(inp.device):
                 celoss_probability_kernel[grid](
-                    inp, tgt, weight, out, label_smoothing, N, C, D
+                    inp,
+                    tgt,
+                    weight,
+                    out,
+                    label_smoothing,
+                    C,
+                    D,
                 )
         elif label_smoothing == 0:
             # target indices
-            w_tgt = torch.zeros(shape, dtype=torch.float32, device=inp.device)
+            w_tgt = torch.zeros([], dtype=torch.float32, device=inp.device)
             with torch.cuda.device(inp.device):
-                celoss_indice_kernel[grid](
-                    inp, tgt, weight, out, w_tgt, ignore_index, N, C, D
+                celoss_indices_kernel[grid](
+                    inp,
+                    tgt,
+                    weight,
+                    out,
+                    w_tgt,
+                    ignore_index,
+                    C,
+                    D,
+                    reduction,
                 )
         else:
-            w_tgt = torch.zeros(shape, dtype=torch.float32, device=inp.device)
+            w_tgt = torch.zeros([], dtype=torch.float32, device=inp.device)
             with torch.cuda.device(inp.device):
-                celoss_indice_smooth_kernel[grid](
-                    inp, tgt, weight, out, w_tgt, ignore_index, label_smoothing, N, C, D
+                celoss_indices_smooth_kernel[grid](
+                    inp,
+                    tgt,
+                    weight,
```

Comment:
Maybe we can optimize weight here. 
The jit function can handle the case that weight is None.(Triton can handle `None` as argument, althought it also involves some kind of overload becase at that case, the type of weight is not a pointer.)

So we do not need to really generate a tensor of ones.

Reply:
done.